test/src/main/java/org/apache/accumulo/test/ScanConsistencyIT.java

```diff
@@ -0,0 +1,492 @@
+            for (Mutation m : tctx.dataTracker.getDeletes()) {
+              writer.addMutation(m);
+              stats.deleted += m.getUpdates().size();
+            }
+          }
+
+        } else {
+
+          List<Mutation> dataAdded = new ArrayList<>();
+
+          int rowsToGenerate = random.nextInt(1000);
+
+          try (var writer = tctx.client.createBatchWriter(tctx.table)) {
+            for (int i = 0; i < rowsToGenerate; i++) {
+              Mutation m = generateMutation(random);
+              writer.addMutation(m);
+              stats.written += m.getUpdates().size();
+              dataAdded.add(m);
+            }
+          }
+
+          // Make the data just written visible to scans. Now that the writer is closed scans should
+          // be able to see the data in the table.
+          tctx.dataTracker.addExpectedData(dataAdded);
+        }
+      }
+
+      return stats;
+    }
+
+    private Mutation generateMutation(Random random) {
+
+      int cols = random.nextInt(100) + 1;
+
+      // the generation counter ensures the row is unique
+      String row = String.format("%016x:%016x", nextLongAbs(random),
+          tctx.generationCounter.getAndIncrement());
+
+      Mutation m = new Mutation(row);
+      for (int i = 0; i < cols; i++) {
+        m.put(String.valueOf(random.nextInt(10)), String.format("%04x", random.nextInt(256 * 256)),
+            "");
+      }
+
+      return m;
+    }
+  }
+
+  private static class TableOpsTask implements Callable<String> {
+    private final TestContext tctx;
+
+    private TableOpsTask(TestContext testContext) {
+      this.tctx = testContext;
+    }
+
+    @SuppressFBWarnings(value = {"PREDICTABLE_RANDOM", "DMI_RANDOM_USED_ONLY_ONCE"},
+        justification = "predictable random is ok for testing")
+    @Override
+    public String call() throws Exception {
+      int numFlushes = 0;
+      int numCompactions = 0;
+      int numSplits = 0;
+
+      Random random = new Random();
+
+      while (tctx.keepRunning.get()) {
+        Thread.sleep(1000);
+
+        int pick = random.nextInt(100);
+
+        if (pick < 10) {
+          // 1 in 10 chance of flushing
+          tctx.client.tableOperations().flush(tctx.table, null, null, random.nextBoolean());
+          numFlushes++;
+        } else if (pick < 15) {
+          // 1 in 20 chance of compacting
+          tctx.client.tableOperations().compact(tctx.table,
+              new CompactionConfig().setFlush(random.nextBoolean()).setWait(random.nextBoolean()));
+          numCompactions++;
+        } else if (pick < 20) {
+          // 1 in 20 chance of splitting
+          int splitsToAdd = random.nextInt(10);
+          TreeSet<Text> splits = new TreeSet<>();
+
+          for (int i = 0; i < splitsToAdd; i++) {
+            splits.add(new Text(String.format("%016x", nextLongAbs(random))));
+          }
+
+          tctx.client.tableOperations().addSplits(tctx.table, splits);
+          numSplits += splitsToAdd;
+        }
```

Comment:
should we add merging?

Reply:
yeah that would be good to add.  maybe bulk import too

Reply:
added merge and other stuff in 7cbc635